src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,57 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final Pair[] elements;
+    private int size;
+
+    public StorageImpl() {
+        elements = new Pair[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size == MAX_SIZE) {
+            throw new ArrayIndexOutOfBoundsException("Array already full");
+        }
+        for (int i = 0; i < elements.length; i++) {
+            if (elements[i] == null) {
+                elements[size] = new Pair<>(key, value);
+                size++;
+                break;
+            } else if (elements[i].key == key
+                    || elements[i].key != null && elements[i].key.equals(key)) {
+                elements[i].value = value;
+                break;
+            }
```

Comment:
I recommend replacing the break statements with return as it directly exits the put method after the new pair is added or the existing key's value is updated. This change would make the code cleaner and more readable by removing the need for the else branch and reducing the method's overall complexity. Using return here is a more conventional approach for such cases in Java.
```suggestion
                return;
            }
            if (keysEqual(key, elements[i].key)) {
                elements[i].value = value;
                return;
            }
```